### PR TITLE
[one-optimize] Remove unnecessary Cast Op

### PR DESCRIPTION
**Pull request by jiwaszki**
This is a draft that introduces `RemoveUnnecessaryCastPass` which removes `Cast` operations with the same input and output types.

For issue https://github.com/Samsung/ONE/issues/13623

### CODE REVIEW

**Comment by jiwaszki** [comment]
@seanshpark @jinevening PR is ready to review.

File: compiler/luci/pass/src/CircleOptimizer.cpp [hunk 1]
```diff
@@ -475,6 +476,10 @@ void CircleOptimizer::optimize(loco::Graph *g) const
   {
     phase.emplace_back(std::make_unique<luci::RemoveUnnecessaryAddPass>());
   }
+  if (_options->query(Options::Algorithm::RemoveUnnecessaryCast))
+  {
+    phase.emplace_back(std::make_unique<luci::RemoveUnnecessaryCastPass>());
+  }
```
Comment (jiwaszki) [inline compiler/luci/pass/src/CircleOptimizer.cpp:482]: Note: update with regards to https://github.com/Samsung/ONE/pull/13800 when it's merged

**Comment by seanshpark** [comment]
LGTM

**Comment by jiwaszki** [comment]
@jinevening sure, I have added you as a reviewer in all of them. Also updated this one to align with all of them.

**Comment by jiwaszki** [comment]
All child PRs has been merged.

**Pull request closed without merging**